### ISSUE TITLE
IntelliJ and Android Studio don't show custom lints

### ISSUE THREAD

**Issue by MartinAKovachev** [invertase/dart_custom_lint#307]
**Describe the bug**
When importing a `- custom_lint` in the analysis_options.yaml file, IntelliJ doesn't show any INFO, WARNING or ERROR message from `custom_lint` in `Dart Analysis` tab nor in the file, in which the  issue occurs.

After running `dart run custom_lint` it shows the ERROR in the terminal, but it's really hard to spot an issue during development.

!The same project is being opened in Visual Studio Code and it shows the ERROR in the file. !

```yaml
include: package:flutter_lints/flutter.yaml

analyzer:
  plugins:
    - custom_lint

custom_lint:
  rules:
    - layer_dependency_rule: true

linter:
  rules:
    avoid_print: true
    prefer_single_quotes: true
    always_use_package_imports: true
    annotate_overrides: true
    use_super_parameters: true
    always_declare_return_types: true

# Additional information about this file can be found at
# https://dart.dev/guides/language/analysis-options
```

flutter: 3.27.2
dart: 3.6.1
custom_lint: ^0.7.1
analyzer: ^7.1.0
analyzer_plugin: ^0.12.0
custom_lint_builder: ^0.7.1

**To Reproduce**

Open a Flutter project in IntelliJ and check if any `custom_lint`s are detected.

**Expected behavior**
`custom_lint`s are shown in the IntelliJ `Dart Analysis` tab and in the file, in which the issue occurs.

@rrousselGit @Rexios80 @matthiasn

**Comment by MartinAKovachev** [comment]
There is a similar bug reported some time ago https://github.com/invertase/dart_custom_lint/issues/191, but the issue still persists.

**Comment by MartinAKovachev** [comment]
@rrousselGit Android Studio and IntelliJ still doesn't work with `custom_lint: ^0.7.2` and `custom_lint_builder: ^0.7.2`.

**Comment by rrousselGit** [comment]
Do you have a reproduction?

**Comment by MartinAKovachev** [comment]
This is what we have in custom_lint_log:
```txt
The request analysis.setContextRoots failed with the following error:
RequestErrorCode.PLUGIN_ERROR
ProcessException: No such file or directory
  Command: flutter pub get
at:
#0      _ProcessImpl._start (dart:io-patch/process_patch.dart:402:33)
#1      Process.start (dart:io-patch/process_patch.dart:38:20)
#2      _runNonInteractiveProcess (dart:io-patch/process_patch.dart:579:18)
#3      Process.run (dart:io-patch/process_patch.dart:49:12)
#4      CustomLintWorkspace.runPubGet (package:custom_lint/src/workspace.dart:755:36)
#5      CustomLintWorkspace.resolvePluginHost (package:custom_lint/src/workspace.dart:748:11)
#6      SocketCustomLintServerToClientChannel._startProcess (package:custom_lint/src/v2/server_to_client_channel.dart:151:24)
#7      SocketCustomLintServerToClientChannel.init (package:custom_lint/src/v2/server_to_client_channel.dart:110:22)
#8      CustomLintServer._maybeSpawnCustomLintPlugin (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:394:25)
<asynchronous suspension>
#9      CustomLintServer._handleAnalysisSetContextRoots.<anonymous closure> (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:341:9)
<asynchronous suspension>
#10     PendingOperation.run (package:custom_lint/src/async_operation.dart:22:14)
<asynchronous suspension>
#11     CustomLintServer._handleRequest (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:173:22)
<asynchronous suspension>
```

**Comment by MartinAKovachev** [comment]
dart pub deps | grep custom_lint
```
├── custom_lint 0.7.2
│   ├── custom_lint_core 0.7.1
│   │   ├── custom_lint_visitor 1.0.0+7.1.0
    ├── custom_lint_builder 0.7.2
    │   ├── custom_lint...
    │   ├── custom_lint_core...
    │   ├── custom_lint_visitor...
```

**Comment by EricSchlichting** [comment]
@rrousselGit  it almost seems like it is running old code. [Platform.resolvedExecutable](https://api.flutter.dev/flutter/dart-io/Platform/resolvedExecutable.html) should be an absolute path according to dart docs. Is there potentially some kind of caching happening of an old package version with the analyzer plugin?

**Comment by rrousselGit** [comment]
I don't see any reason why old code would be executed.

**Comment by chika3742** [comment]
Same issue.
I tried:
- Invalidate caches and restart
- deleting lockfile
- `flutter pub cache clean`
- deleting `~/.dartServer`
- restarting Mac

Finally I created a symlink in `.pub-cache` as a temporary measure.
```
rm -rf ~/.pub-cache/hosted/pub.dev/custom_lint-0.7.0
ln -s ~/.pub-cache/hosted/pub.dev/custom_lint-0.7.2 ~/.pub-cache/hosted/pub.dev/custom_lint-0.7.0
```

**Comment by iamwaseemsh** [comment]
Please use the following versions to resolve the issue.
  custom_lint: ^0.6.10

  analyzer: ^6.11.0
  analyzer_plugin: ^0.11.3
  custom_lint_builder: ^0.6.10

**Comment by pavkoccino** [comment]
I am confused. It seems for some people it works at certain setups and for some doesn't. I am currently running:

In the linting package:
```dart
  analyzer: ^6.6.0
  analyzer_plugin: ^0.11.0
  custom_lint_builder: ^0.6.8
``` 

In the project where I use it:
```dart
  custom_lint: ^0.6.8
``` 

Works in VSCode, but does not work in Android studio. Same symptoms. IDE does not show it, but if I run `dart run custom_lint`, it captures the error correctly.

**Comment by ronnnnn** [comment]
I found the issue is caused by not updating custom_lint version on `packages/custom_lint/tools/analyzer_plugin/pubspec.yaml`.
This may be fixed by #309.

Workaround until new version is published is updating your pubspec.yaml and pubspeck.lock in `~/.dartServer/.plugin_manager/.../analyzer_plugin/` like below.

pubspec.yaml

```yaml
name: custom_lint_analyzer_plugin_loader
description: This pubspec determines the version of the analyzer plugin to load.
version: 0.7.0
publish_to: none

environment:
  sdk: ">=3.0.0 <4.0.0"

dependencies:
  custom_lint: 0.7.2
```

pubspeck.lock

```
# Generated by pub
# See https://dart.dev/tools/pub/glossary#lockfile
packages:
  # ...
  custom_lint:
    dependency: "direct main"
    description:
      name: custom_lint
      sha256: "7e6a73e1722ad90b48f77918994c17a5b177b0869804ea3c4ce9c9199de019f6"
      url: "https://pub.dev"
    source: hosted
    version: "0.7.2"
  custom_lint_core:
    dependency: transitive
    description:
      name: custom_lint_core
      sha256: "6dcee8a017181941c51a110da7e267c1d104dc74bec8862eeb8c85b5c8759a9e"
      url: "https://pub.dev"
    source: hosted
    version: "0.7.1"
  custom_lint_visitor:
    dependency: transitive
    description:
      name: custom_lint_visitor
      sha256: "14df0760dfa81b7b0c398c876045f4e4a343eb2c9d200c66163671dd3e337c1b"
      url: "https://pub.dev"
    source: hosted
    version: "1.0.0+7.1.0"
```

**Comment by rrousselGit** [comment]
Looks like my versioning broke and no-longer bumps the tool folder. I'll investigate, but in the meantime I'll just release a new version by hand

**Comment by rrousselGit** [comment]
Should be good now

**Comment by ronnnnn** [comment]
@rrousselGit 
Thanks. It works as expected.

**Comment by Nobler** [comment]
I'm facing a similar issue in android studio: the lint rule works in existed files immedately, but works in new files after some  miniutes or longer. In vscode, it works in new files after some seconds, also not immedately.

Because my flutter version is 3.10.0, cannot use the latest version 0.7.x because it using analyze 7.0.
Is that a same issue? Any hotfixs can be applyed for 0.6.x? 

Thanks.
